presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcPageSourceFactory.java

```diff
@@ -149,7 +150,7 @@ public static OrcPageSource createOrcPageSource(
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(sessionUser, path, configuration);
             long size = fileSystem.getFileStatus(path).getLen();
             FSDataInputStream inputStream = fileSystem.open(path);
-            orcDataSource = new HdfsOrcDataSource(path.toString(), size, maxMergeDistance, maxBufferSize, streamBufferSize, inputStream, stats);
+            orcDataSource = new HdfsOrcDataSource(new OrcDataSourceId(path.toString()), size, maxMergeDistance, maxBufferSize, streamBufferSize, inputStream, stats);
```

Comment:
Yes, the goal here is to make it easy to identify the corrupt file in Hive or Raptor that is failing the query. This is one of those things you find by being the admin of a large production system (and being annoyed at spending an hour tracking down the file).

For both data sources, they'll be URL or path.